src/main/java/com/box/sdk/BoxResourceIterable.java

```diff
@@ -0,0 +1,158 @@
+package com.box.sdk;
+
+import java.net.URL;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+import com.eclipsesource.json.JsonArray;
+import com.eclipsesource.json.JsonObject;
+
+/**
+ * Common implementation for paging support.
+ *
+ * @author Stanislav Dvorscak
+ *
+ * @param <T>
+ *            type of iterated entity
+ */
+public abstract class BoxResourceIterable<T> implements Iterable<T> {
```

Comment:
You can use `JSONIterator` which already has a lot of this functionality. See [`BoxGroupIterator`](https://github.com/box/box-java-sdk/blob/master/src/main/java/com/box/sdk/BoxGroupIterator.java) as an example of a paging iterator.